manifests/charts/gateways/istio-egress/templates/deployment.yaml

```diff
@@ -46,6 +46,13 @@ spec:
 {{ toYaml $gateway.podAnnotations | indent 8 }}
 {{ end }}
     spec:
+{{- if not $gateway.runAsRoot }}
+      securityContext:
```

Comment:
Please consider also setting the other fields in this `securityContext` to tighten the security, [like we do for sidecar proxies](https://github.com/kubernetes/enhancements/pull/1003#issuecomment-619087019):
```
        allowPrivilegeEscalation: false
        capabilities:
          drop:
          - ALL
        privileged: false
        readOnlyRootFilesystem: true
        runAsGroup: 1337
        fsGroup: 1337
        runAsNonRoot: true
        runAsUser: 1337
```